lib/elixir/lib/stream.ex

```diff
@@ -1106,8 +1106,12 @@ defmodule Stream do
       [{1, :a, "foo"}, {2, :b, "bar"}, {3, :c, "baz"}]
 
   """
-  @spec zip([Enumerable.t()]) :: Enumerable.t()
-  def zip(enumerables) when is_list(enumerables) do
+  @spec zip([Enumerable.t()] | %Stream{}) :: Enumerable.t()
+  def zip(%Stream{} = enumerables), do: zip_fun(enumerables)
+
+  def zip(enumerables) when is_list(enumerables), do: zip_fun(enumerables)
+
+  defp zip_fun(enumerables) do
```

Comment:
This is a a pretty bad name but I can't think of a better one 😄 . I thought about calling it `_zip` but I think it seems to clash with the convention of unimported functions. Any advice on this would be great!